core/leader/leader.go

```diff
@@ -50,10 +44,45 @@ func Run(db *sql.DB, addr string, lead func(context.Context)) {
 	}
 	log.Messagef(ctx, "Using leaderKey: %q", l.key)
 
-	update(ctx, l)
-	for range time.Tick(5 * time.Second) {
-		update(ctx, l)
+	var leadCtx context.Context
+	var cancel func()
+	for leader := range leadershipChanges(ctx, l) {
```

Comment:
Like if something else sent a value over the channel? Right now, the `leader` value is ferried directly from the source of truth, which is nice. Moving the `leader` value into this loop would make it clearly strictly alternate, but introduce the possibility that it might be on the wrong state.

What do you think about something like

```
var leadCtx context.Context
var cancel func()
for {
    <-becameLeaderCh
    log.Messagef(ctx, "I am the core leader")
    leadCtx, cancel = context.WithCancel(ctx)
    l.lead(leadCtx)
    isLeading.Store(true)

    <-demotedCh
    log.Messagef(ctx, "No longer core leader")
    cancel()
    isLeading.Store(false)
}
```